os/net/ipv6/uip-nd6.c

```diff
@@ -202,6 +204,11 @@ ns_input(void)
 #endif /* UIP_CONF_IPV6_CHECKS */
     switch (ND6_OPT_HDR_BUF(nd6_opt_offset)->type) {
     case UIP_ND6_OPT_SLLAO:
+      if(uip_l3_icmp_hdr_len + nd6_opt_offset +
+         UIP_ND6_OPT_DATA_OFFSET + UIP_LLADDR_LEN > uip_len) {
+        LOG_ERR("Insufficient space for NS SLLAO option\n");
```

Comment:
Maybe this comment should be Insufficient data for ... since it is a read and not a write (write needs space and read needs data).

Reply:
Yes, it would be better to say that. I've changed it now.